app/packs/src/components/collection_management/GenericSegmentsTabs.js

```diff
@@ -0,0 +1,266 @@
+
+    this.onStoreChange = this.onStoreChange.bind(this);
+    this.handleSave = this.handleSave.bind(this);
+    this.handleSelectNav = this.handleSelectNav.bind(this);
+    this.clickedOnBack = this.clickedOnBack.bind(this);
+  }
+
+  componentDidMount() {
+    CollectionStore.listen(this.onStoreChange);
+    CollectionActions.fetchUnsharedCollectionRoots();
+  }
+
+  componentWillUnmount() {
+    CollectionStore.unlisten(this.onStoreChange);
+  }
+
+  onStoreChange(state) {
+    let children = state.unsharedRoots.length > 0 ? state.unsharedRoots : [{}];
+
+    this.setState({
+      tree: {
+        label: 'My Collections',
+        id: -1,
+        children: children
+      }
+    });
+  }
+
+  onClickNode(node) {
+    this.setState({ currentCollection: node });
+    this.handleSelectModalOptions(this.state.selectModal);
+  }
+
+  getArrayFromLayout = (layout, availableTabs) => {
+    const layoutKeys = Object.keys(layout);
+    const enabled = availableTabs.filter(val => layoutKeys.includes(val));
+    const leftover = availableTabs.filter(val => !layoutKeys.includes(val));
+    const visible = [];
+    const hidden = [];
+
+    enabled.forEach((key) => {
+      const order = layout[key];
+      if (order < 0) { hidden[Math.abs(order)] = key; }
+      if (order > 0) { visible[order] = key; }
+    });
+
+    leftover.forEach(key => hidden.push(key));
+
+    let first = null;
+    if (visible.length === 0) {
+      first = hidden.filter(n => n !== undefined)[0];
+      if (first) {
+        visible.push(first);
+      }
+    }
+    if (hidden.length === 0) {
+      hidden.push('hidden');
+    }
+
+    return {
+      visible: Immutable.List(visible.filter(n => n !== undefined)),
+      hidden: Immutable.List(hidden.filter(n => (n !== undefined && n !== first)))
+    };
+  };
+
+  handleChange(tree) {
+    this.setState({
+      tree: tree
+    });
+  }
+
+  onClicked(node) {
+    this.setState({ currentCollection: node });
+    this.onClickNode(node);
+  }
+
+  handleModalOptions(showModal) {
+    this.setState({ showModal: !showModal });
+  }
+
+  handleSelectModalOptions(selectModal) {
+    this.setState({ selectModal: !selectModal });
+  }
+
+  selectCurrentTab(key) {
+    const tabTitles = {
+      0: 'sample',
+      1: 'reaction',
+      2: 'wellplate',
+      3: 'screen'
+    };
+    this.setState({ currentTab: tabTitles[key] });
+    return tabTitles[key];
+  }
+
+  clickedOnBack() {
+    this.handleModalOptions(this.state.showModal);
+    this.handleSelectModalOptions(this.state.selectModal);
+  }
+
+  handleSelectNav(eventKey) {
+    const currentTab = this.selectCurrentTab(eventKey);
+    let layout = {};
+    const node = this.state.currentCollection;
+    if (!isEmpty(node.tabs_segment[currentTab])){
+      layout = node.tabs_segment[currentTab];
+    } else {
+      layout = {
+        analyses: -1,
+        literature: 3,
+        properties: 1,
+        qc_curation: 2,
+        references: -3,
+        results: -2
+      };
```

Comment:
the default layout should come from profile, not hard coding.
not all elements have the same tabs.
for example, reaction's tabs are different from samples.
if the layout comes from profile, then it includes all tabs already (including all segments)
![screenshot-localhost_3000-2022 05 06-17_11_57](https://user-images.githubusercontent.com/70375322/167162247-3465ae63-a742-41d5-84cd-2620eeae2aab.png)
![screenshot-localhost_3000-2022 05 06-17_12_13](https://user-images.githubusercontent.com/70375322/167162253-75f273c2-e478-46ae-aa5b-64eeab38bf7e.png)